safe_transaction_service/history/tests/test_views_v2.py

```diff
@@ -145,3 +156,316 @@ def test_safe_collectibles_paginated(self):
+            )
+            self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+            # Create delegate
+            self.assertEqual(SafeContractDelegate.objects.count(), 0)
+            chain_id = self.ethereum_client.get_chain_id()
+            hash_to_sign = DelegateSignatureHelperV2.calculate_hash(
+                delegate.address, chain_id, False
+            )
+            data["signature"] = delegator.signHash(hash_to_sign)["signature"].hex()
+            response = self.client.post(url, format="json", data=data)
+            self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+            safe_contract_delegate = SafeContractDelegate.objects.get()
+            self.assertEqual(safe_contract_delegate.delegate, delegate.address)
+            self.assertEqual(safe_contract_delegate.delegator, delegator.address)
+            self.assertEqual(safe_contract_delegate.label, label)
+            self.assertEqual(safe_contract_delegate.safe_contract_id, safe_address)
+
+            # Update label
+            label = "Jimmy McGill"
+            data["label"] = label
+            response = self.client.post(url, format="json", data=data)
+            self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+            self.assertEqual(SafeContractDelegate.objects.count(), 1)
+            safe_contract_delegate = SafeContractDelegate.objects.get()
+            self.assertEqual(safe_contract_delegate.label, label)
+
+        # Create delegate without a Safe
+        hash_to_sign = DelegateSignatureHelperV2.calculate_hash(
+            delegate.address, chain_id, False
+        )
+        data = {
+            "label": "Kim Wexler",
+            "delegate": delegate.address,
+            "delegator": delegator.address,
+            "signature": delegator.signHash(hash_to_sign)["signature"].hex(),
+        }
+        response = self.client.post(url, format="json", data=data)
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        self.assertEqual(SafeContractDelegate.objects.count(), 2)
+
+        # Test not internal server error on contract signature
+        signature = signature_to_bytes(0, int(delegator.address, 16), 65) + HexBytes(
+            "0" * 65
+        )
+        data["signature"] = signature.hex()
+        response = self.client.post(url, format="json", data=data)
+        self.assertIn(
+            f"Signature of type=CONTRACT_SIGNATURE for signer={delegator.address} is not valid",
+            response.data["non_field_errors"][0],
+        )
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_delegates_get(self):
+        url = reverse("v2:history:delegates")
+        response = self.client.get(url, format="json")
+        self.assertEqual(response.data[0], "At least one query param must be provided")
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
```

Comment:
Could you include a test for not found delegate address?

Reply:
Added.